src/Core/Security/Voter/PermissionVoter.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Forumify\Core\Security\Voter;
+
+use Forumify\Core\Entity\User;
+use Forumify\Plugin\Entity\Permission;
+use Forumify\Core\Repository\PermissionRepository;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+class PermissionVoter extends Voter
+{
+    private PermissionRepository $permissionRepository;
+
+    public function __construct(PermissionRepository $permissionRepository)
+    {
+        $this->permissionRepository = $permissionRepository;
+    }
+
+    protected function supports(string $attribute, $subject): bool
+    {
+        return preg_match('/^[a-z]+\.[a-z]+\.[a-z]+\.[a-z]+$/', $attribute) === 1;
```

Comment:
this regex won't match anymore if we allow Nth level of depth. I'm inclined to say just `return true` here since forumify uses the `affirmative` method of voting. So it *shouldn't* interfere with other voters afaik.